gax/src/main/java/com/google/api/gax/batching/v2/BatcherImpl.java

```diff
@@ -120,35 +141,38 @@ private void sendBatch() {
     if (currentOpenBatch == null) {
       return;
     }
-    final Batch<ElementT, ElementResultT, RequestT, ResponseT> accumulatedBatch = currentOpenBatch;
-    currentOpenBatch = null;
+    synchronized (elementLock) {
+      final Batch<ElementT, ElementResultT, RequestT, ResponseT> accumulatedBatch =
+          currentOpenBatch;
+      currentOpenBatch = null;
```

Comment:
I think only the swap needs to be synchronized.
Adding the callbacks risks invoking them under lock, which we don't want to do

Reply:
Updated with the same.